### PR TITLE
Update the MySQL version used by our Docker images

### LINKED ISSUE #11948: Update the MySQL version used in the Docker lite images
The MySQL version we are using in our `vitess/lite` images is pretty old (`8.0.23`) we should upgrade it to a more recent and stable version: `8.0.31`.

- [ ] Find the implications of this change. Run some upgrade/downgrade tests between v15 and v16 on docker. and also with the operator.
- [ ] Use those findings to warns users in the release notes.
- [ ] We also need to change the default docker image we publish from 5.7 to 8.0.

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This PR upgrades the default MySQL version used by the `vitess/lite:latest` and `vitess/lite:mysql80` images.

- `vitess/lite:latest`: `5.7` -> `8.0.31`
- `vitess/lite:mysql80`: `8.0.23` -> `8.0.31`

## Related Issue(s)

- Fixes #11948

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: t/main.go [hunk 1]
```diff
@@ -0,0 +1 @@
+package t
```
Comment (deepthi) [inline t/main.go:1]: I'm guessing this file was added by mistake? you probably need to delete it.

File: examples/operator/302_new_shards_new_images.yaml [hunk 2]
```diff
@@ -0,0 +1,185 @@
+apiVersion: planetscale.com/v2
```
Comment (deepthi) [inline examples/operator/302_new_shards_new_images.yaml:1]: Why do we need this new file?

File: docker/lite/Dockerfile.mysql80 [hunk 3]
```diff
@@ -35,6 +35,8 @@ RUN make install PREFIX=/vt/install
 # Start over and build the final image.
 FROM debian:buster-slim
 
+LABEL com.vitessio.mysql-version="8.0.31"
```
Comment (deepthi) [inline docker/lite/Dockerfile.mysql80:38]: What is the idea of adding this label?

File: docker/base/Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2019 The Vitess Authors.
+# Copyright 2023 The Vitess Authors.
```
Comment (deepthi) [inline docker/base/Dockerfile:1]: We don't change the copyright year on edit.

File: docker/base/Dockerfile.percona57 [hunk 5]
```diff
@@ -1,3 +1,17 @@
+# Copyright 2023 The Vitess Authors.
```
Comment (deepthi) [inline docker/base/Dockerfile.percona57:1]: 👍

**Review by deepthi — COMMENTED** [review]
The changes mostly look good.
I'm assuming this is still draft because of a dependency with changes to vitess-operator.

File: examples/operator/302_new_shards_new_images.yaml (hunk 2 above)
Comment (frouioui) [reply examples/operator/302_new_shards_new_images.yaml:1]: This was for test purposes

**Review by GuptaManan100 — APPROVED** [review]

File: doc/releasenotes/16_0_0_summary.md [hunk 6]
```diff
@@ -56,6 +56,10 @@ The flag is available in these components:
 
 ### Breaking Changes
 
+#### Default MySQL version on Docker
+
+The default MySQL version use by our `vitess/lite:latest` image is going from `5.7` to `8.0`. Additionally, the default version of the `vitess/lite:mysql80` image goes from `8.0.23` to `8.0.31`.
```
Comment (mattlord) [inline doc/releasenotes/16_0_0_summary.md:61]: `MySQL version used` and `the default patch version of the`... we can make touch-ups like that as part of the release process though.

**Review by mattlord — APPROVED** [review]

File: doc/releasenotes/16_0_0_summary.md (hunk 6 above)
Comment (frouioui) [reply doc/releasenotes/16_0_0_summary.md:61]: Follow up PR: https://github.com/vitessio/vitess/pull/12222

**Pull request merged**